core/src/main/scala/cats/syntax/apply.scala

```diff
@@ -14,6 +14,34 @@ trait ApplySyntax extends TupleSemigroupalSyntax {
     new ApplyOps(fa)
 }
 
+final class IfAOps[F[_]](val fa: F[Boolean]) extends AnyVal {
+
+  /**
+   * A conditional lifted into the `F` context.
+   *
+   * Example:
+   * {{{
+   * scala> import cats.implicits._
+   *
+   * scala> val b1: Option[Boolean] = Some(true)
+   * scala> val asInt1: Option[Int] = b1.ifA(Some(1), Some(0))
+   * scala> asInt1.get
+   * res0: Int = 1
+   *
+   * scala> val b2: Option[Boolean] = Some(false)
+   * scala> val asInt2: Option[Int] = b2.ifA(Some(1), Some(0))
+   * scala> asInt2.get
+   * res1: Int = 0
+   *
+   * scala> val b3: Option[Boolean] = Some(true)
+   * scala> val asInt3: Option[Int] = b3.ifA(Some(1), None)
+   * asInt2: Option[Int] = None
+   *
+   * }}}
```

Comment:
Unfortunately we can't add any methods to type classses without breaking binary compatibility. So for now, we have to settle adding to a new syntax trait. Sorry I am on my phone and couldn't point you to an example.

Reply:
@kailuowang If I understand you correctly, we can not add the method to the `trait Apply[F[_]]` (the type-class) but can only leave it as part of a syntactic extension `implicit class`?

Reply:
sorry for the delayed response. Yes, that's the idea. You already updated the PR, we also need to create a new `ApplySyntaxBinCompat` trait to include an implicit conversion from `F[Boolean]` to `IfAOps`, then there are a couple of more steps, 
1.  [here](https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/syntax/all.scala) we need to add a new trait `AllSyntaxBinCompat4` to extends the new `ApplySyntaxBinCompat`,
 let the `AllSyntaxBinCompat` at the top extend this `AllSyntaxBinCompat4`
2. [here] https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/implicits.scala we also need to let the `implicits` object extends this new `AllSyntaxBinCompat4`
3. then we can probably add a doctest here that `import cats.implicits._` to validate that it's all hooked up. 

Sorry about the gymnastics you have to go through to overcome the Scala 2.11 binary compatibility restrictions.